system/filesystem.py

```diff
@@ -41,6 +41,12 @@
     description:
     - If yes, allows to create new filesystem on devices that already has filesystem.
     required: false
+  growfs:
+    choices: [ "yes", "no" ]
+    default: "no"
+    description:
+    - If yes, if the block device and filessytem size differ, grow the filesystem into the space. This DOES NOT support shrinking. Will only grow if the FS is mounted.
```

Comment:
The comment about mounted FS should really be a warning about XFS specifically. It doesn't actually check if it's mounted or not, so the warning would make more sense.

I don't think you can express shrinking in a playbook. Consider this playbook structure.

Create VG
Create Lvol
Create Filesystem with growfs

So, now you change the Lvol size, and it grows. All good. 

So what if you reduce the lvol size? You would reduce the block size BEFORE you shrink the FS. So you would actually need to re-organise to:

Create VG
Create Filesystem with growfs
Create Lvol

Now it will shrink. But at the same time you have removed the ability to grow unless you ran the play twice ... 

I think that shrinking will probably need more consideration, and would likely need direct integration with the lvol module.